spdm_dump/spdm/spdm_dump_spdm.c

```diff
@@ -1041,14 +1041,14 @@ void dump_spdm_certificate(const void *buffer, size_t buffer_size)
                     printf("Fail to write out_req_cert_chain\n");
                 }
             }
-            if (m_requester_cert_chain_buffer == NULL ||
-                m_requester_cert_chain_buffer_size == 0) {
-                m_requester_cert_chain_buffer =
+            if (m_requester_cert_chain_buffer[spdm_response->header.param1] == NULL ||
```

Comment:
what if param1 is 0xf (provisioned pub key)?

Reply:
For get_certificate and certificate_rsp command, the param1 will always be between 0 and 7. 

For key_exchange and key_exchange_rsp command, SlotID shall be 0xFF or 0xF  if the public key was provisioned previously. And I have added the check.

The TH1 is not related to the cert_chain buffer, when the certificate is not uesd.
![image](https://user-images.githubusercontent.com/87060980/191229535-d3f36eb7-4632-4a1d-b28a-92caa52872c1.png)

Reply:
Thanks. I have added check at the dump_certificate function beginning.